src/main/scala/io/shiftleft/fuzzyc2cpg/passes/cfgcreation/CfgCreator.scala

```diff
@@ -0,0 +1,404 @@
+package io.shiftleft.fuzzyc2cpg.passes.cfgcreation
+
+import io.shiftleft.codepropertygraph.generated.{EdgeTypes, Operators, nodes}
+import io.shiftleft.codepropertygraph.generated.nodes.CfgNode
+import io.shiftleft.fuzzyc2cpg.passes.cfgcreation.Cfg.CfgEdgeType
+import io.shiftleft.passes.DiffGraph
+import io.shiftleft.semanticcpg.language._
+
+/**
+  * Translation of abstract syntax trees into control flow graphs
+  *
+  * The problem of translating an abstract syntax tree into a corresponding
+  * control flow graph can be formulated as recursive problem in which
+  * sub trees of the syntax tree are translated and their corresponding
+  * control flow graphs and are connected according to the control flow
+  * semantics of the root node.
+  * For example, consider the abstract syntax tree for an if-statement:
+  *
+  *               (  if )
+  *              /       \
+  *          (x < 10)  (x += 1)
+  *            / \       / \
+  *           x  10     x   1
+  *
+  * This tree can be translated into a control flow graph, by translating
+  * the sub tree rooted in `x < 10` and that of `x += 1` and connecting
+  * their control flow graphs according to the semantics of `if`:
+  *
+  *            [x < 10]----
+  *               |t     f|
+  *            [x +=1 ]   |
+  *               |
+  * The semantics of if dictate that the first sub tree to the left
+  * is a condition, which is connected to the CFG of the second sub
+  * tree - the body of the if statement - via a control flow edge with
+  * the `true` label (indicated in the illustration by `t`), and to the CFG
+  * of any follow-up code via a `false` edge (indicated by `f`).
+  *
+  * A problem that becomes immediately apparent in the illustration is that
+  * the result of translating a sub tree may leave us with edges for which
+  * a source node is known but the destination not depends on parent or
```

Comment:
```suggestion
a source node is known but the destination does not depend on parent or
```

Reply:
This one seems wrong, let me verify and push a fix.